src/Weather/Skysight/APIQueue.hpp

```diff
@@ -24,12 +24,14 @@ Copyright_License {
 #define WEATHER_SKYSIGHTAPI_QUEUE_HPP
 
 #include "Request.hpp"
+#include "CDFDecoder.hpp"
 #include "Metrics.hpp"
 #include "Event/Timer.hpp"
 #include <vector>
 
 class SkysightAPIQueue final: public Timer {
   std::vector<std::unique_ptr<SkysightAsyncRequest>> request_queue;
+  std::vector<std::unique_ptr<CDFDecoder>> decode_queue;
```

Comment:
Why a vector of unique_ptrs? Why not just a list?

Reply:
I give up. Could you guide me how to convert a vector of unique_ptrs to a list?